ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala

```diff
@@ -98,22 +99,26 @@ object KeyValueCommitting {
 
     // Look at what kind of submission this is...
     submission.getPayloadCase match {
-      case DamlSubmission.PayloadCase.ARCHIVE =>
-        val archive = submission.getArchive
-        val key = DamlStateKey.newBuilder.setPackageId(archive.getHash).build
+      case DamlSubmission.PayloadCase.PACKAGE_UPLOAD_ENTRY =>
+        val archives = submission.getPackageUploadEntry.getArchivesList.asScala
         // TODO(JM): We're duplicating the archive data. This way we don't have an indirection
         // to fetch by packageId or when building [[Update]], and we don't have to declare
         // the log entry containing the archive as an input (which would be messy).
-        logger.trace(
-          s"processSubmission[entryId=${prettyEntryId(entryId)}]: Package ${archive.getHash} committed.")
+
+        // TODO(MZ): Produce a DamlPackageUploadEntry that filters out pre-existing packages
```

Comment:
this could be done by depending on the state entries for each of the packages and then filtering out the ones that exist. it'll read all the data for them from the database though, so we might want to consider extending the interface a bit.